tutorials/abap-environment-shell-plugin-enrich/abap-environment-shell-plugin-enrich.md

```diff
@@ -0,0 +1,411 @@
+                };
+                var i18nModel = new ResourceModel({
+                    bundleName: "zpluginxxx.i18n.i18n"
+                });
+                oRenderer.setModel(i18nModel, "i18n");
+                oRenderer.addUserPreferencesEntry(oEntry);
+            }
+        });
+      }
+    );
+```
+> If you have chosen different names for your project and for your Fragment View in the previous steps, you will need to adjust the code above: replace any instance of `zpluginxxx` with the id of your project, and replace `BusinessUserFragment` with the name of your View.
+
+### Preview your SAPUI5 Application
+Your enhanced SAPUI5 Application can be previewed before it is deployed. Using the preview functionality is good practice before deploying an application, to make sure that the UI looks as intended and the application's functionalities work as desired.
+
+1. Right click on your project folder and select **Preview Application**.
+
+    ![Preview application](Preview_app_1_new.png)
+
+2. Select **start fiori-run**. This will prompt a new terminal window which will run the **start fiori-run** command. This can take up to a few seconds. Once it is done, a preview of your application will open up in a new browser window.
+
+    ![Preview application - start fiori run](Preview_app_2_new.png)
+
+3. In the preview of your application, click on the user icon in the top-right corner and open the **Settings**. Your shell plug-in is listed in the user's settings dialog. User-related information is displayed according to the layout you defined using the Fragment View.
+
+    ![Application preview - settings](Preview_app_3.png)
+    
+    ![Application preview - shell plug-in](Preview_app_4.png)
+
+### Deploy your Enhanced Shell Plug-in
+You can now re-deploy your SAPUI5 Application with an enhanced shell plug-in to your SAP BTP ABAP Environment.
+
+1. In Business Application Studio, right click on the project folder and select **Open in Integrated Terminal**. This will open a new terminal window.
+
+    ![Open terminal window](open_terminal.png)
+
+2. In the terminal window, run the command: `npm run deploy`. The deployment process will begin. You will be prompted to confirm the deployment by typing `Y`. Once the SAPUI5 Application is successfully deployed to your SAP BTP ABAP Environment, the terminal will return the status:
+`Deployment Successful`.
+>The deployment process can take up to a few minutes.
+
+### Test your Shell Plug-in
+After a successful deployment, your shell plug-in will be updated in the system. Login to your SAP BTP ABAP Environment with a Business User with the necessary Business Role (see previous tutorial in this series). Open your application by clicking on the application tile. The plug-in can be accessed via the user settings dialog. The plug-in returns data related to your Business User and a list of Business Roles that are assigned to your Business User.
+
+![Final plugin](final_plugin.png)
+
+### Test Yourself
```

Comment:
This is the tenth step.  If you want the validation to appear in this step, you need to modify the VR file so that it states [VALIDATE_10] instead of [VALIDATE_9}.